inc_internal/ziti_enroll.h

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright (c) 2019-2020 NetFoundry, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+
+#ifndef ZITI_SDK_ENROLL_H
+#define ZITI_SDK_ENROLL_H
+
+#define MBEDTLS_PLATFORM_C
+#include <mbedtls/platform.h>
+#include <mbedtls/entropy.h>
+#include <mbedtls/error.h>
+#include <mbedtls/pk.h>
+#include <mbedtls/ecdsa.h>
+#include <mbedtls/rsa.h>
+#include <mbedtls/error.h>
+#include <mbedtls/entropy.h>
+#include <mbedtls/ctr_drbg.h>
+#include <mbedtls/x509_csr.h>
+#include "ziti_model.h"
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct enroll_cfg_s {
+
+    nf_enroll_cb external_enroll_cb;
+
+    ziti_enrollment_jwt *zej;
+
+    char *CA;
+
+    char *PrivateKey;
+    mbedtls_pk_context pk_context;
+
+    unsigned char x509_CSR[4096];
+    mbedtls_x509write_csr x509_csr;
```

Comment:
I haven't gotten to the point where these fields are used yet, but it my hunch is that it would be nice if their names were differentiated by more than case.

Reply:
I changed the names of these fields to be `x509_csr_pem` and `x509_csr_ctx` to better indicate what they are.